torch/csrc/jit/export.cpp

```diff
@@ -303,16 +305,15 @@ void EncoderBase::EncodeBlock(
       EncodeBlock(false_g, node->blocks()[1]);
     }
   }
-  auto num_initializers = initializers.size();
-  AT_ASSERT(block->inputs().size() >= num_initializers);
-  size_t inputs_count = block->inputs().size() - num_initializers;
-  for (auto& tensor : initializers) {
-    // TODO: stop using positions to determine which initializers
-    // match to which inputs
-    std::string name = graph_proto->input(inputs_count++).name();
+  AT_ASSERT(block->inputs().size() >= initializers.size());
+  // Copying initializers to into an (ordered) stdd::map so that
+  // initializers always gets written into ONNX graph in the same 
+  // order. Mostly done for testing purpose.
+  for (auto& name_tensor_pair : 
+        std::map<std::string, at::Tensor>(initializers.begin(), initializers.end())) {
```

Comment:
Which tests will fail? We probably can change it. This hack seems not a good way to solve the problem

Reply:
Updated based on discussion to std::map.